tests/dbus/test-commissioner

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+#
+#  Copyright (c) 2020, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+# Test build and run otbr dbus commissioner
+#
+
+set -euxo pipefail
+
+readonly OTBR_DBUS_SERVER_CONF=otbr-test-agent.conf
+readonly OT_CTL="${CMAKE_BINARY_DIR}"/third_party/openthread/repo/src/posix/ot-ctl
+
+on_exit()
+{
+    local status=$?
+
+    sudo killall otbr-agent || true
+    sudo killall expect || true
+    sudo killall ot-cli-ftd || true
+    sudo rm "/etc/dbus-1/system.d/${OTBR_DBUS_SERVER_CONF}" || true
+
+    return "${status}"
+}
+
+main()
+{
+    sudo rm -rf tmp
+    sudo install -m 644 "${CMAKE_BINARY_DIR}"/src/agent/otbr-agent.conf /etc/dbus-1/system.d/"${OTBR_DBUS_SERVER_CONF}"
+    sudo service dbus reload
+    trap on_exit EXIT
+    sudo "${CMAKE_BINARY_DIR}"/src/agent/otbr-agent -d7 -v -I wpan0 "spinel+hdlc+forkpty://$(command -v ot-rcp)?forkpty-arg=1" &
+    sleep 2
+    sudo "${OT_CTL}" factoryreset
+    sudo "${OT_CTL}" panid 0x1234
+    sudo "${OT_CTL}" ifconfig up
+    sudo "${OT_CTL}" thread start
+    sleep 5
+    expect <<EOF &
+spawn ot-cli-ftd 2
+sleep 5
+send "ifconfig up\r\n"
+expect "Done"
+send "joiner start ABCDEF\r\n"
+expect "Done"
+wait
```

Comment:
expect join success?

Reply:
We've already asserted that in the commissioner test cpp.